web3.js/test/connection.test.ts

```diff
@@ -2002,6 +2002,27 @@ describe('Connection', () => {
       await connection.removeRootChangeListener(subscriptionId);
     });
 
+    it('logs notification', async () => {
+      let listener: number | undefined;
+      const owner = new Account();
+      const [logsRes, ctx] = await new Promise(resolve => {
+        listener = connection.onLogs('all', (logs, ctx) => {
```

Comment:
You'll want to add a commitment to avoid the test timeout here. By default the "finalized" commitment is used so the logs won't be delivered for awhile. I think "processed" would be good here

Reply:
The test still intermittently fails with `processed`. Investigating.

Reply:
Only the browser tests have the problem (with or without the commitment param).

Reply:
I believe the problem is a race condition between setting up the subscription in `onLogs` and subsequently executing the the transaction to trigger the logs.

If the transaction to trigger the log gets executed before the subscription is setup, the event listener is never fired and so the promise is never resolved. 

I've added a `sleep` to work around this race condition.

Reply:
Ah makes sense, thanks for investigating!